INT insertion at aggregate switch

Sanjhana-Jayagopal opened the pull request:
#### What does this PR do?
Fixes #169 
Fixes #187 
- Adds INT header insertion at the aggregate
- Adds test cases to validate the processing of INT and non-INT packets at the ingress of the aggregate switch.
#### Do you have any concerns with this PR?
No
#### How can the reviewer verify this PR?
Run CI
#### Any background context you want to provide? 
No
#### Screenshots or logs (if appropriate)
#### Questions:
- Have you connected this PR to the issue it resolves? Yes
- Does the documentation need an update? No
- Does this add new dependencies? No
- Have you added unit or functional tests for this PR?

Commit:
INT insertion at aggregate switch

Commit:
INT insertion at aggregate switch

Commit:
Merge remote-tracking branch 'origin/int-insert-agg' into int-insert-agg

Sanjhana-Jayagopal commented:
Run lab trial build

spisarski reviewed (changes requested):
While fixing the tofino aggregate tests, please make the two minor formatting changes.

Commit:
Fixed formatting

Commit:
Fixed tofino aggregate issues

Sanjhana-Jayagopal commented:
- Added a match key to aggregate's 'add_switch_id_t' to fix issues with Tofino aggregate tests.
- Removed 'device_id' as an action parameter from aggregate's 'data_inspection_t' table.
- Send 'add_switch_id_t' table entry when instantiating a switch.

Sanjhana-Jayagopal commented:
Run Tofino aggregate build

spisarski commented on trans_sec/p4runtime_lib/bmv2.py:
```diff
@@ -292,6 +312,20 @@ def add_data_inspection(self, dev_id, dev_mac):
+        table_entry = self.p4info_helper.build_table_entry(
+            table_name='{}.add_switch_id_t'.format(self.p4_ingress),
+            match_fields={
+                'hdr.udp.dst_port': 0x022b
```
Can you use the value consts.UDP_INT_DST_PORT here?

spisarski commented on playbooks/scenarios/full/packet-flood.yml:
```diff
@@ -37,7 +37,7 @@
     send_packet_count: 150
     min_received_packet_count: 99
-    max_received_packet_count: 150
+    max_received_packet_count: "{{ send_packet_count if send_protocol == 'UDP' else send_packet_count|int*3 }}"
```
Can't do this. You will be masking an issue with dropping packets that currently is not occurring.
Please replace with either 150 or "{{ send_packet_count }}" which I actually like better.

spisarski commented on playbooks/scenarios/full/packet-flood.yml:
```diff
@@ -66,7 +66,7 @@
     send_packet_count: 150
     min_received_packet_count: 99
-    max_received_packet_count: 150
+    max_received_packet_count: "{{ send_packet_count if send_protocol == 'UDP' else send_packet_count|int*3 }}"
```
Can't do this. You will be masking an issue with dropping packets that currently is not occurring.
Please replace with either 150 or "{{ send_packet_count }}" which I actually like better.

spisarski commented on playbooks/scenarios/full/packet-flood.yml:
```diff
@@ -95,7 +95,7 @@
     send_packet_count: 150
     min_received_packet_count: 99
-    max_received_packet_count: 150
+    max_received_packet_count: "{{ send_packet_count if send_protocol == 'UDP' else send_packet_count|int*3 }}"
```
Can't do this. You will be masking an issue with dropping packets that currently is not occurring.
Please replace with either 150 or "{{ send_packet_count }}" which I actually like better.

spisarski commented on playbooks/scenarios/lab_trial/data-inspection.yml:
```diff
@@ -0,0 +1,101 @@
+        name: "{{ 'data_inspect_packet' }}"
+        match_fields:
+          - key: hdr.udp_int.dst_port
+            value: "{{ '0x022b' }}"
```
Try and see if the value 555 (without quotes) will work here

spisarski commented on playbooks/scenarios/lab_trial/data_inspection_basic.yml:
```diff
@@ -0,0 +1,72 @@
+    send_src_port: "{{ '%s' % range(2000, 10000) | random(seed=now|int + 10) }}"
+    send_packet_count: "{{ '%s' % 100 | random(seed=now|int + 11) }}"
+    inspection_data: "{{ sr1_data_inspection_int | default(None) }}"
+    max_received_packet_count: "{{ send_packet_count if send_protocol == 'UDP' else send_packet_count|int*2 }}"
```
This isn't as bad as what we have in packet-flood.yml, but I believe this is masking an issue with TCP acks.

spisarski reviewed (changes requested):
I think we've gotten our TCP ack issue back.

spisarski committed:
Changes to TCP tests on expected packets received

spisarski committed:
Fixed poor refactoring
